code/API_definitions/qod-api.yaml

```diff
@@ -92,7 +92,24 @@ paths:
       tags:
         - QoS Sessions
       summary: Creates a new session
-      description: Create QoS Session to manage latency/throughput priorities
+      description: |
+        Create QoS Session to manage latency/throughput priorities
+
+        If the qosStatus in the API response is "AVAILABLE" and a notification callback is provided the client will receive in addition to the response a
+        `QOS_STATUS_CHANGED` event notification with `qosStatus` as `AVAILABLE`.
```

Comment:
From a developer's point of view, I'm unsure how it can be possible for the QoD-Application to directly respond with 'AVAILABLE' after creating the session. 
According to the [3GPP standard for using '3gpp-as-session-with-qos'](https://www.3gpp.org/ftp/specs/archive/OpenAPI/Rel-16/TS29122_AsSessionWithQoS.yaml), the response of a created subscription does not immediately provide information about ,e.g., 'SUCCESSFUL_RESOURCES_ALLOCATION.' 
This information is typically delivered through a callback, as far as I know.

So, in my understanding, after the session creation is successful, the qosStatus can only be 'REQUESTED.'

What do you think, @hdamker?

Reply:
I agree that in general this would not be implemented as a none blocking call, so most, if not all implementations would not use `SUCCESSFUL_RESOURCES_ALLOCATION`.

Including this as an optional return code won't break anything. Since never returning it allowed.

Reply:
Here we are not creating a subscription but a QoS session with implicit notifications. We are always assuming that session is synchronously created (201) even if QoS may not be immediately available. The problem with this approach is that complicates billing for QoS sessions, as only the part of the session with QoS available should be billed.

Reply:
@maxl2287 The API should be technology agnostic. While an implementation against the 3GPP SCEF/NEF standard will - as you described - always return initially 'REQUESTED' (which the API allows), implementations of qosProfiles against other technologies might be immediately 'AVAILABLE' after the creation of the QoS session resource (e.g. if using pre-allocated resources or just prioritisation).

Reply:
> We are always assuming that session is synchronously created (201) even if QoS may not be immediately available. The problem with this approach is that complicates billing for QoS sessions, as only the part of the session with QoS available should be billed.

@jlurien My understanding it here a little bit different: there is the "QoD session resource" which will be created synchronously, and there is the actual network QoS session which get's 'available' and 'unavailable'. The ability to bill this correctly is one of the reason to create the respective events as described within this PR.

Reply:
@hdamker My understanding for what we have right now is the same. My comment is a reflection about whether our current approach is what consumers will expect or understand, and the implications that the current model may have on QoS session billing, after some feedback I recently received. Client will ask for a QoS session of 1h which counts from the creation (201) until expiration or deletion, but QoS is only effectively provided between status=AVAILABLE and status=UNAVAILABLE. But this is probably a discussion for another issue.